src/main/java/quotify_app/data_access/PredictionClient.java

```diff
@@ -0,0 +1,66 @@
+package quotify_app.data_access;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import quotify_app.data_access.exceptions.PredictionClientException;
+import quotify_app.entities.PredictionRequest;
+
+/**
+ * A client for interacting with the Housing Price Prediction API.
+ */
+public class PredictionClient {
+    private static final String PREDICT_ENDPOINT = "https://housing-prediction-api-62e3f0c37c7d.herokuapp.com/predict";
```

Comment:
This one is indeed only used here and for the specific functionality this client provides. If we had more endpoints, it might make sense to separate them, but yeah I just left it here because its the only one.